mvc/models/model-binding.rst

```diff
@@ -1,8 +1,113 @@
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/retro-sample.txt
+   :language: html
+
+The above code quickly becomes repetitive, cumbersome, and error prone to write, even when it's just a few fields. Real world apps often have many fields combined with complex structures. Admittedly, this kind of mapping code is a bit tedious to write as well. 
+
+How model binding works
+-----------------------
+When ASP.NET MVC receives an incoming HTTP request, it routes it to a specific action method of a controller. It determines which action method to run based on what is in the route data, then it binds values from the HTTP request to that action method's parameters. For example, consider the following URL:
+
+`http://contoso.com/movies/edit/2`
+
+The default model binder compares the sections of the URL with the pattern defined in the ``RouteValues`` collection. Put a breakpoint in any action method to see the contents of the ``RouteData`` object at run time in a debug window, as illustrated here: 
+
+.. image:: model-binding/_static/routevalues-in-watch-window.png
+
+Since the  routing pattern looks like this, ``{controller=Home}/{action=Index}/{id?}``, the route ``movies/edit/2`` maps to the ``Movies`` controller, and its ``Edit`` action method. It also accepts an optional ``id`` of 2 as the image shows here:
+
+.. image:: model-binding/_static/model-binding.png
+
+This is due to MVC's use of conventions. So the code for the action method should look something like this: 
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/Controllers/MoviesController.cs
+   :language: c#
+   :lines: 63-76  
+   
+.. note:: The strings in the URL are not case sensitive.
+
+Routing knows which action method to use, so it checks for model binding attributes applied to the action method's arguments. If it finds a model binding attribute it will use the data source specified by the attribute. Otherwise, for each argument (a primitive type for now, more on complex types later), it looks in its data sources for something with a matching name. In the above example the only argument is an integer named `id`, so that is what the binder will look for. Below is a list of the data sources in the order that the model binder looks through them:
+ 
+#. ``Request.Form``: These are form values that go in the HTTP request using the POST method.
```

Comment:
Instead of type and property names let's use descriptive text:
Request.Form -> Form data
RouteData.Values -> Route data
Request.QueryStrings -> Query strings
etc.